pkg/network/tracer/connection/kprobe/config.go

```diff
@@ -74,27 +74,35 @@ func enabledProbes(c *config.Config, runtimeTracer bool) (map[probes.ProbeFuncNa
 		enableProbe(enabled, probes.InetBindRet)
 		enableProbe(enabled, probes.UDPSendPage)
 		enableProbe(enabled, probes.UDPSendPageReturn)
-
+		if kv >= kv470 || runtimeTracer {
+			enableProbe(enabled, probes.UDPRecvMsg)
+		} else if kv > kv410 {
+			enableProbe(enabled, probes.UDPRecvMsgPre470)
+		} else {
+			enableProbe(enabled, probes.UDPRecvMsgPre410)
+		}
+		enableProbe(enabled, selectVersionBasedProbe(runtimeTracer, kv, probes.UDPRecvMsgReturn, probes.UDPRecvMsgReturnPre470, kv470))
 		if c.CollectIPv6Conns {
 			enableProbe(enabled, selectVersionBasedProbe(runtimeTracer, kv, probes.IP6MakeSkb, probes.IP6MakeSkbPre470, kv470))
+			if kv >= kv470 || runtimeTracer {
+				enableProbe(enabled, probes.UDPv6RecvMsg)
+			} else if kv > kv410 {
+				enableProbe(enabled, probes.UDPv6RecvMsgPre470)
+			} else {
+				enableProbe(enabled, probes.UDPv6RecvMsgPre410)
+			}
+			enableProbe(enabled, selectVersionBasedProbe(runtimeTracer, kv, probes.UDPv6RecvMsgReturn, probes.UDPv6RecvMsgReturnPre470, kv470))
 			enableProbe(enabled, probes.IP6MakeSkbReturn)
 			enableProbe(enabled, probes.Inet6Bind)
 			enableProbe(enabled, probes.Inet6BindRet)
 		}
 
-		if runtimeTracer {
+		if runtimeTracer || kv >= kv470 {
```

Comment:
Don't you always want to run this check now?

Reply:
The offset guessing for sk_buff does not work for older kernels. I tried it on 4.4 and it did not work, so I've kept the current limitation of >= 4.7.